src/main/java/ChatBot.java

```diff
@@ -0,0 +1,109 @@
+import command.ByeCommand;
+import command.CheckCommand;
+import command.DeadlineCommand;
+import command.DeleteCommand;
+import command.EventCommand;
+import command.FindCommand;
+import command.ListCommand;
+import command.TodoCommand;
+import command.UncheckCommand;
+
+import task.TaskManager;
+
+import ui.WelcomeUI;
+
+import util.DukeException;
+import util.FileManager;
+
+import java.util.Scanner;
+
+/**
+ * Represents the main logic flow of the chatbot.
+ */
+public class ChatBot {
+    private Scanner input;
+    private TaskManager taskManager;
+    private String[] inputArr;
+    private FileManager fileManager;
+
+    /**
+     * Initialises the Scanner to recieve user input
+     * and the WelcomeUI to be displayed upon start up.
+     * @param input
+     * @param welcomeUI
+     */
+    public ChatBot(Scanner input, WelcomeUI welcomeUI) {
+        System.out.println(welcomeUI);
+        this.input = input;
+        this.taskManager = new TaskManager();
+        this.fileManager = new FileManager();
+        this.fileManager.loadDataToArrayList(this.taskManager);
+    }
+
+    /**
+     * Driver method to run the chatbot.
+     */
+    public void run() {
+        loop: while (input.hasNextLine()) {
+            //remove leading and trailing whitespaces
+            String ip = input.nextLine().trim();
+
+            this.inputArr = ip.split(" ", 2);
+            int len = inputArr.length;
+
+            if (ip.isBlank()) {
+                System.out.println("No command given, please give me one!");
+            }
+
+            String command = inputArr[0];
+            try {
+                switch (command) {
+                case "":
+                    continue;
+                case "list":
+                    ListCommand lc = new ListCommand(taskManager);
+                    lc.executeCommand();
+                    break;
+                case "bye":
+                    ByeCommand bc = new ByeCommand();
+                    bc.executeCommand();
+                    fileManager.saveTasksToFile(taskManager);
+                    break loop;
```

Comment:
Since you override the `executeCommand()` method in all child classes of `Command`, perhaps you can utilise polymorphism and call the method outside the `switch case` block?

Perhaps you could also consider exiting the program within the `byeCommand` 's implementation of `executeCommand()`?